torch/csrc/jit/codegen/cuda/arith.cpp

```diff
@@ -251,11 +271,23 @@ DataType getOutputType(BinaryOpType op_type, Val* v1, Val* v2) {
   const bool all_integer_input =
       isIntegralType(v1_dtype) && isIntegralType(v2_dtype);
 
+  // Combine categories
+  const auto v1_cat = getCategory(v1);
+  const auto v2_cat = getCategory(v2);
+  if (v1_cat != v2_cat) {
+    const DataType higher = v1_cat > v2_cat ? v1_dtype : v2_dtype;
+    const DataType lower = v1_cat > v2_cat ? v2_dtype : v1_dtype;
+    const DataType common_type = getCommonType(higher, lower);
+    v1_dtype = common_type;
+    v2_dtype = common_type;
+  }
+
   if (isIntegerOp(op_type) || (alsoBooleanOperator(op_type) && integer_input)) {
     // If integer op or maybe bool op with integer inputs meaning binary op
     if (integer_input && all_integer_input) {
       return promote_type(v1_dtype, v2_dtype);
     } else if (integer_input && !all_integer_input) {
+      // Question: this doesn't look like what eager mode does
```

Comment:
@csarofeen  I actually have a question on this logic since eager mode doesn't seem to do this, for example: 
```python
a=torch.tensor([1],dtype=torch.float,device="cuda")
b=torch.tensor([1],dtype=torch.long,device="cuda")
c=a>>b # c.dtype == torch.float
```
I was wondering if it is safe to change this part.

Reply:
I don't think we should any bitwise operators on float tensors at this time. I'm uncertain who should use them or why.

Reply:
Sure. Added assertion here so we just fallback if any user happens to use this kind of op.